app/domain/Application.scala

```diff
@@ -376,6 +380,10 @@ case class Application(id: String,
   }
 
   def hasLockedSubscriptions = deployedTo.isProduction && state.name != State.TESTING
+
+  def findCollaboratorByHash(teamMemberHash: String) : Option[Collaborator] = {
```

Comment:
What about hashing clashes ?

Reply:
This was just refactoring the existing behaviour into a method. It already worked like this. I expect we won't get clashes (as it's only scoped to emails for a single app, and there's never very many!)